src/transformers/pipelines/visual_question_answering.py

```diff
@@ -0,0 +1,165 @@
+
+    def _sanitize_parameters(self, top_k=None, padding=None, truncation=None, **kwargs):
+        preprocess_params, postprocess_params = {}, {}
+        if padding is not None:
+            preprocess_params["padding"] = padding
+        if truncation is not None:
+            preprocess_params["truncation"] = truncation
+        if top_k is not None:
+            postprocess_params["top_k"] = top_k
+        return preprocess_params, {}, postprocess_params
+
+    def __call__(self, *args, **kwargs):
+        r"""
+        Answers open-ended questions about images. The pipeline accepts several types of inputs which are detailed
+        below:
+
+        - `pipeline(image=image, question=question)`
+        - `pipeline(image=[image], question=[question])`
+        - `pipeline(image=[image, image], question=[question])`
+        - `pipeline(image=[image], question=[question, question])`
+        - `pipeline(image=[image], question=[question, question])`
+        - `pipeline({"image": image, "question": question})`
+        - `pipeline([{"image": image, "question": question}])`
+        - `pipeline([{"image": image, "question": question}, {"image": image, "question": question}])`
+
+        Args:
+            image (`str`, `List[str]`, `PIL.Image` or `List[PIL.Image]`):
+                The pipeline handles three types of images:
+
+                - A string containing a http link pointing to an image
+                - A string containing a local path to an image
+                - An image loaded in PIL directly
+
+                The pipeline accepts either a single image or a batch of images. If given a single image, it can be
+                broadcasted to multiple questions.
+            question (`str`, `List[str]`):
+                The question(s) asked. If given a single question, it can be broadcasted to multiple images.
+            top_k (`int`, *optional*, defaults to 5):
+                The number of top labels that will be returned by the pipeline. If the provided number is higher than
+                the number of labels available in the model configuration, it will default to the number of labels.
+        Return:
+            A dictionary or a list of dictionaries containing result. The dictionaries contain the following keys:
+
+            - **label** (`str`) -- The label identified by the model.
+            - **score** (`int`) -- The score attributed by the model for that label.
+        """
+
+        pipeline_inputs = self._args_parser(*args, **kwargs)
+        results = super().__call__(pipeline_inputs, **kwargs)
+        return results
+
+    def preprocess(self, pipeline_input, padding=False, truncation=False):
+        image = load_image(pipeline_input["image"])
+        model_inputs = self.tokenizer(
+            pipeline_input["question"], return_tensors=self.framework, padding=padding, truncation=truncation
+        )
+        image_features = self.feature_extractor(images=image, return_tensors=self.framework)
```

Comment:
Wondering if we can't leverage `ViltProcessor` here, which applies the tokenizer and feature extractor to prepare everything for the model in one go.

Reply:
I thought about using `ViltProcessor` but I think using the `tokenizer` and the `feature_extractor` separately is a more principled approach because these are existing arguments to the `pipeline` in general.

Reply:
Ok for me